File: 351003/Naskevich/src/models/post.py
from dataclasses import dataclass, field

from src.models.post_state import PostState


@dataclass(kw_only=True)
class Post:
    """Обсуждение к твиту. В Cassandra: PRIMARY KEY ((tweet_id), id); tweet_id — ключ партиции."""

    id: int = field(default=0, init=False)
    tweet_id: int
    content: str
    state: PostState = PostState.PENDING

File: 351004/Zalivako/DistributedComputing/src/DiscussionService/Models/Post.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DiscussionService.Models
{

    public enum PostState
    {
        PENDING,
        APPROVE,
        DECLINE
    }

    public class Post
    {
        [BsonId] // основной ключ
        [BsonRepresentation(BsonType.Int64)]
        public long Id { get; set; }

        [BsonElement("newsId")]
        public long NewsId { get; set; }

        [BsonElement("content")]
        public string Content { get; set; } = string.Empty;

        [BsonElement("country")]
        public string Country { get; set; } = string.Empty;

        [BsonElement("state")]
        public PostState State { get; set; } = PostState.PENDING;
    }
}
